# 08 Submitting sequences to Sequence Read Archive (SRA)
This document will provide an overview of how to submit your sequences to the Sequence Read Archive service hosted by NCBI.  

Objectives:
- Understand the purpose of submitting sequences to NCBI SRA
- Understand the basic submission process  

## What is SRA? 
Sequence Read Archive (SRA) is a database service hosted by the National Center for Biotechnology Infomration, or NCBI, which itself is hosted by the National Library of Medicine, which is part of the National Institutes of Health, which is part of the US Department of Health and Human Services :weary:. If you ever see a reference to an "accession number" in a paper, this usually means that the project has been registered in SRA and has a unique identifier associated with all of the samples from that project. There are a few different databases that NCBI host for different purposes, but the main one that is relevant for our work with 16S amplicon sequencing is SRA.  

SRA is a database that accepts all kinds of sequencing data. SRA takes direct submissions of data and also indexes data from other major international databases. SRA is completely free to access and provides some (not super user friendly tbh) ways of accessing data through their [website](https://www.ncbi.nlm.nih.gov/sra/) or command line tools. Using the data hosted on SRA, you can revisit data that was previously generated by others to inform your own research or analyze it using new tools.  

Submitting your raw sequencing data to SRA is important for a few key reasons. First, SRA is a safe location for long-term data storage provided the US government doesn't completely collapse. While you should definitely still store your sequences in the Wells Sharepoint and wherever you are working on Quest, SRA is more reliable than either of those two locations and can be accessed freely and easily. Second, submitting your raw data improves the credibility of your work. Having raw data easily accessible is basically the expectation in environmental microbial ecology work and, in manuscripts, reviewers will typically comment if you do not provide an NCBI accession number or other public database where your raw sequences are available. Finally, having your raw sequencing data accessible to others is important for reproducibility and advancing the field more broadly. Providing raw sequences (and your analysis scripts) can provide a way for others to validate your work. Furthermore, sequencing data collected from one reactor or experiment can be useful in a variety of contexts besides the immediate goals of the project or your PhD research as a whole that you may not have thought of.  

## Anatomy of the NCBI database
The most confusing part of using NCBI databases is understanding how each piece of the database structure fits together. I will describe the most relevant information for 16S data in this document.  

The largest container of data is a **BioProject**, which is assigned a unique accession number. The BioProject consists of a title and brief description, and the BioProject page will host links to every piece of data associated with that particular project. *This is the most common way that sequencing data is shared in manuscripts using an accession number.* To access a BioProject page, visit the main NCBI webpage - https://www.ncbi.nlm.nih.gov - then use the search bar to look up a specific accession. BioProject accession numbers often start with PR in the prefix. You can see an example BioProject page at this accession number: PRJNA897867.  

The next container down is called a **BioSample**. Each BioSample represents a unique sample that was collected from a particular study system at a particular time and date. For example, if you sequenced DNA from 10 sampling dates from one bioreactor, you would end up with 10 BioSamples in the BioProject. If you sequenced DNA from 10 sampling dates from two bioreactors, you would have 20 BioSamples. Note that BioSamples do NOT represent individual sequencing runs, just unique metadata. For example, if you sequenced two duplicates from the same date, these would be listed under the same BioSample. Furthermore, if you did two types of sequencing on the same sample, like 16S rRNA amplicons and shotgun metagenomic sequencing, these would still be housed under the same BioSample. BioSample accession numbers often start with SAMN in the prefix. You can see an example BioSample page at this accession number: SAMN31582570.  

The smallest container of data is the **SRA Experiment**. Each SRA Experiment represents the unique sequencing data type associated with a BioSample. So looping back to our last example, a 16S rRNA amplicon sequencing run and a shotgun metagenomic sequencing run would be separate SRA Experiments associated with the same BioSample. If you just did 16S sequencing, you would have one BioSample and one associated SRA Experiment. You can go back and add more SRA Experiments or BioSamples after you've set up a BioProject, so no worries if this changes over time! SRA Experiment accession numbers often start with SRX in the prefix. Here is an example of an SRA Experiment: SRX18156033. 

## How to submit data to SRA
You should submit your sequences to SRA as soon as possible after getting the raw data back, mostly so you don't forget to do it and so you don't have to scramble to do it as you're finishing up a manuscript or project. You can set the data to release to the public after a set date or release it immediately upon submission. The submission process is a bit convoluted, but I've listed the general steps below (as of early 2024). Note that this is for raw 16S amplicon sequences and specific steps will be different if you are submitting other types of data.  

1. Get your sample `fastq.gz` files, sample names, and metadata in order. A HUGE quirk and annoyance of SRA is that the sample names *across your entire account and all BioProjects* must be unique. *You cannot have sample names that are the same across BioProjects.* So don't start your first SRA submission with sample names like S1, S2, S3, etc. I would recommend the following:  
    1. On Quest, copy your raw reads to a new folder. Don't do any of this on your original raw reads lol. 
    2. Set your working directory to your new folder. Get rid of any extra files in the directory so there are just `fastq.gz` files, then get a list of the filenames using `ls -l`. Copy these names into Excel and delete all the random stuff that isn't the sample names. 
    3. Make a new column with whatever you want the new sample name to look like. I would recommend distinguishing letters or numbers that represent your particular project and then the unique letter and number identifier of each sample from your original sample names. So for example, I recently submitted 16S rRNA amplicon sequences for a project at O'Brien Water Reclamation Plant. I changed sample names into this pattern: `16S_OB_[sample_name]`, 16S stands for 16S PCR products, OB stands for O'Brien, and sample_name was the original sample names I used in the original submission to the sequencing center. Here is how the filename conversion ended up being for one sample: `1961-27-SBR2-14_S364_L001_R1_001.fastq.gz` -> `16S_OB_SBR2_R1.fastq.gz`. Remember that you will have a forward and reverse read for each sample name denoted by R1 and R2.   
    4. Format the file so the first column is the original filename and the second column is the new filename. Do not add a header row. 
    5. Save the file as a .csv. You may have to convert the line endings from CRLF (often the Windows and Microsoft product default) to LF, which you can do in VSCode or other text editors.
    6. Upload this file to Quest in the directory you saved your reads in. 
    7. Run the following code *in the Quest terminal* in the directory where your reads are to change the filenames. If you get an error or something weird happens (like the line ending issue), just delete the directory and start over by copying the raw reads again.  
    `sed 's/"//g' files.csv | while IFS=, read orig new; do mv "$orig" "$new"; done`  
    8. You are done for now but will need to log back on to Quest to submit the data through the command line. Keep a list of the new sample names handy in an Excel or text file as well. Remember that the unique sample names will NOT include R1 or R2. The easiest way for me to do this in Excel was making a new column with the last part of the filename chopped off with `=left()`, then using `=unique()` on that new column to get the sample names. 


2. Go to https://submit.ncbi.nlm.nih.gov and sign in with an ORCID. If you don't have an ORCID, you will need to create one here: https://orcid.org/register. There are other sign in options but *I highly recommend using an ORCID because it will persist after you leave Northwestern and is useful for other stuff*.  

3. Go to the "My submissions" tab and select "Sequence Read Archive" under the new submissions options, then hit "New submission"  

![My submissions tab](/docs/images/my_submissions.png)  

![New submission page](/docs/images/sra_new.png)  

4. Fill out the contact info page and use the department address for the contact information. 

5. On the general info page, select "no" for the BioProject and BioSample options. You will be assigned a BioProject accession number and will be able to populate your BioSample information.  

![General info page](/docs/images/general_info.png)  


6. Fill out the project information. You will need a title and brief description. In the description, I usually mention the project location, super brief purpose of the project, and type of sequencing. You can update this description later if you end up adding more types of data or expanding the project.  

7. Under BioSample type, select the "Packages for metagenome submitters" tab then select "Metagenome or environmental". You can also select the MIMARKS survey package but the metadata requirements are a bit more of a hassle to deal with than the "Metagenome or environmental" category so I tend not to choose it.  

8. On the next page, you will need to fill in the BioSample metadata information. You can either use an in-browser table editor or download an Excel file and fill out the metadata. It's way easier to edit the Excel file. Use the sample names that you decided on in the first step, then fill in the rest of the information. You can find an example submission file [here](/resources/Metagenome.environmental.1.0.xlsx). There are some columns that are mandatory but many are not so be sure to read the descriptions. Also make sure to use whatever updated Excel file is available on the SRA submission portal, do not directly work out of my file since it may be out of date. Upload the completed spreadsheet back to the portal. 

9. The next step is basically the same thing but for the SRA Experiments. Again, download the template Excel file and fill it out. Check out [this example](/docs/SRA_metadata.xlsx) for some guidance but make sure you work out of a fresh copy downloaded from the submission portal. For paired end reads, you will use the forward read in the filename column and reverse read in the filename2 column. Upload the completed spreadsheet back to the portal. You will get an error message if your sample names aren't consistent between both Excel files you just filled out.

10. Upload your data using the FTP connection. Select "Upload using FTP". You will need to log into Quest and use the terminal to do the rest of the steps. 
    1. Set your current directory to the folder where you have your renamed reads in on Quest. All of your reads should be in the same folder. 
    2. Connect to the server via FTP and follow the basic instructions given on the submission portal. You will type this into the Quest terminal `ftp [whatever URL is on the submission portal]`, then enter the username and password provided in the submission portal. You will now be logged into the FTP connection, and your working directory on Quest is still retained.
    3. Follow the instructions on the portal to navigate to your personal upload folder then make a new directory. You have to follow those instructions exactly or your data upload won't work.
    4. Type `prompt` into the FTP active terminal and hit enter. You want the interactive mode turned OFF so you don't have to approve each file copying action manually. When I last did this, the interactive mode was on by default and using `prompt` turned it off, so double check this before going to the next step. 
    5. Copy the raw reads with this command `mput *.fastq.gz`.
    6. Once the copying action is done, log out of the FTP connection using `exit`.
    7. Go back to the SRA submission portal and see if you can access your files. You will have to wait about 10-15 minutes for the files to be visible. 
    8. Hit submit. Any major errors with filename discrepancies should be recognized at this point. 

11. Review your SRA submission then submit! The rest of the process is automated and you will get an email when everything is good to go. 

As mentioned earlier, this process will be different with different types of sequencing data. Also, you don't have to rename the raw reads to make the upload work properly, but I find this to be more consistent than using the default filenames from the sequencing center. Either way, you will have to map your metadata to the new, unique filenames you've created for the SRA submission. 
